src/Storages/HDFS/ReadBufferFromHDFS.cpp

```diff
@@ -137,7 +137,7 @@ std::optional<size_t> ReadBufferFromHDFS::getTotalSize()
 bool ReadBufferFromHDFS::nextImpl()
 {
     impl->position() = impl->buffer().begin() + offset();
-    auto result = impl->next();
```

Comment:
Not a good fix, `!hasPendingData()` assertion is a reason of `impl->position()` not being updated correctly in the line above, though it seems correct and some time ago I checked hdfs worked fine with debug mode. Better to understand what is wrong with `impl->position()` now.

Reply:
@kssenii  https://github.com/ClickHouse/ClickHouse/pull/24885 PLS check this PR. I think that bug is imported by it. I I It is not good because two level buffer here.

Reply:
@kssenii ReadBufferFromHDFS is just a thin wrapper after this PR: https://github.com/ClickHouse/ClickHouse/pull/24885/files#

Reply:
@kssenii Fix.